Mathlib/Tactic/Ring/RingNF.lean

```diff
@@ -238,7 +243,8 @@ elab (name := ring1NF) "ring1_nf" tk:"!"? cfg:(config ?) : tactic => do
 
 /--
 Tactic for evaluating expressions in *commutative* (semi)rings, allowing for variables in the
-exponent.
+exponent. It uses `ring1` and `ring_nf` in succession so, if `ring1` fails, `ring_nf` will
+not be called.
```

Comment:
Speaking for myself, I'm not sure this addition actually clarifies anything. (I have a reasonable intuition for what `ring` does, but don't have a clue about `ring1`.) I'm not blocking this PR, but please think about whether this is the most useful thing to show users?

Reply:
the user-level explanation of `ring1`  is: it's `ring`, but it fails instead of trying and suggesting `ring_nf` if the goal is not appropriate for `ring`.

Reply:
Changed accordingly and added example. I think it's good to have, as I myself was puzzled for a long time by the `ring_nf` suggestion.